src/rtls.rs

```diff
@@ -95,14 +95,20 @@ impl TlsConnector for Arc<rustls::ClientConfig> {
         dns_name: &str,
         mut tcp_stream: TcpStream,
     ) -> Result<Box<dyn HttpsStream>, Error> {
-        let sni =
-            rustls::ServerName::try_from(dns_name).map_err(|e| ErrorKind::Dns.new().src(e))?;
+        let sni = rustls::ServerName::try_from(dns_name).map_err(|e| {
+            ErrorKind::Dns
+                .msg(format!("dns server name failed ({})", dns_name))
```

Comment:
I like that we're adding the name that failed here! How about:

```suggestion
                .msg(format!("parsing '{}'", dns_name))
```

So it would be something like:

"Dns Failed: parsing 'ex%mple.com': invalid dns name"

In general there's a question of whether we want to say:

"failed to do X: Y" vs

"doing X: Y"

Right now we say "failed to do X"; I think "doing X" is shorter and composes better.

As another example, below we could say:

"Network Error: creating TLS client: failed to get random bytes"

By the way, we should make a unittest that exercises as many of the error paths as we reasonably can, and asserts the current Display output. That would let us see all the error messages as rendered and make sure they make sense and are as nice as we can make them.